IFC_Adapter/Methods/Convert/Physical/FromIFC/Floor.cs

```diff
@@ -43,10 +45,19 @@ public static Floor FloorFromIfc(this IIfcSlab element, IfcSettings settings)
                 return null;
             }
 
-            settings = settings.DefaultIfNull();
+            IfcObject ifcObject = element as IfcObject;
+            if (ifcObject == null)
+            {
+                BH.Engine.Reflection.Compute.RecordError("The IFC element could not be read due to an internal error.");
```

Comment:
I'd change the error text to something like
```suggestion
                BH.Engine.Reflection.Compute.RecordError($"The provided IFC element of type {element.GetType().Name} could not be converted to {nameof(IfcObject)}.");
```